dlib/matrix/matrix_fft.h

```diff
@@ -9,836 +9,135 @@
 #include "../algs.h"
 
 #ifdef DLIB_USE_MKL_FFT
-#include <mkl_dfti.h>
-#endif
-
-// No using FFTW until it becomes thread safe!
-#if 0
-#ifdef DLIB_USE_FFTW
-#include <fftw3.h>
-#endif // DLIB_USE_FFTW
+#include "mkl_fft.h"
+#define impl_fft    mkl_fft
+#define impl_fftr   mkl_fftr
+#define impl_fftri  mkl_fftri
+#else
+#include "kiss_fft.h"
+#define impl_fft    kiss_fft
+#define impl_fftr   kiss_fftr
+#define impl_fftri  kiss_fftri
```

Comment:
Don't use `#defines`.  It makes the code harder to follow and might clash with symbols in other people's code.  It's fine to just give these functions the same names in both the mkl_fft.h and kiss_fft.h files.  That's what most dlib code that does this kind of switching does.  Then this kind of name switching here is not needed.

Reply:
So what names should i give everyone? ```impl_fft```, ```impl_fftr``` and ```impl_fftri``` ?

Reply:
Yeah `impl_*` is fine.  Although probably better to put them in a namespace, like `impl::fft()`.